### PR TITLE
Support moe w8a8

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch/kernels/cuda/w8a8_fused_moe.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+    for k in range(0, tl.cdiv(K, BLOCK_SIZE_K)):
+        a = tl.load(a_ptrs,
+                    mask=mask_sid[:, None] &
+                    (offs_k[None, :] < K - k * BLOCK_SIZE_K),
+                    other=0.0)
+        b = tl.load(b_ptrs,
+                    mask=offs_k[:, None] < K - k * BLOCK_SIZE_K,
+                    other=0.0)
+        accumulator += tl.dot(a, b)
```
Comment (RunningLeon) [inline lmdeploy/pytorch/kernels/cuda/w8a8_fused_moe.py:144]: may change to `accumulator = tl.dot(a, b, accumulator)` and add accumulator_dtype for `fp8` inference

File: lmdeploy/pytorch/kernels/cuda/w8a8_fused_moe.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    if M <= 0:
+        return
+
+    num_pid_m = tl.cdiv(M_NP2, BLOCK_SIZE_M)
+    num_pid_n = tl.cdiv(N, BLOCK_SIZE_N)
+
+    if GROUP_SIZE_M == 1:
+        pid_m = pid % num_pid_m
+        pid_n = pid // num_pid_m
+        # pid_m = pid // num_pid_n
```
Comment (RunningLeon) [inline lmdeploy/pytorch/kernels/cuda/w8a8_fused_moe.py:99]: may remove these two unused lines.

**Review by RunningLeon — APPROVED** [review]
LGTM

File: lmdeploy/pytorch/models/mixtral.py [hunk 3]
```diff
@@ -182,12 +185,10 @@ def __init__(self,
                                        device=device)
 
         # build attention layer norm
-        self.post_attention_layernorm = RMSNorm(
-            config.hidden_size,
-            config.rms_norm_eps,
-            quant_config=quantization_config,
-            dtype=dtype,
-            device=device)
+        self.post_attention_layernorm = RMSNorm(config.hidden_size,
```
[the remark this answers is not in the document]
Comment (AllentDan) [reply lmdeploy/pytorch/models/mixtral.py:188]: I saw quantization_config not removed in other moe models. Shall we remove them as well?

**Review by AllentDan — APPROVED** [review]
Mixtral-W8A8 tested OK

**Pull request merged**